src/AfterLogin/QuizComponent/MixAndMatch/QuizFirstPageMixAndMatch.js

```diff
@@ -0,0 +1,126 @@
+import CloseIcon from "@mui/icons-material/Close";
+import {
+  FormGroup,
+  FormControlLabel,
+  Checkbox,
+  Button,
+  Card,
+  CircularProgress,
+} from "@mui/material";
+import { Link } from "react-router-dom";
+//Take the user data from App.js state
+
+//Component let user choose which decks to include in the quiz
+export default function QuizFirstPageMixAndMatch(props) {
+  const userDecks = props.userDecks;
+
+  //handle change for user to choose/unchoose decks
+  const handleChange = (e, deck) => {
+    const isChecked = e.target.checked;
+    if (isChecked) {
+      props.setDecks((prev) => [...prev, deck]);
+    } else {
+      props.setDecks((prev) => {
+        const deckID = e.target.value;
+        const index = prev.findIndex((deck) => deck.deckID === deckID);
+        const reduced = prev.toSpliced(index, 1);
+        return reduced;
+      });
+    }
+  };
+
+  //component show the decks option
+  const selection =
+    userDecks && userDecks.length ? (
```

Comment:
```suggestion
    userDecks && userDecks.length ? (
```

these 2 checks are redundant, as an empty array will just return `[]` when `.map()` runs. `[]` is truthy as well

Reply:
I would rather store the result in a variable, then conditional render based on the length